### ISSUE TITLE
User-defined range hyperparameter of ```StronglyEntanglingLayers``` not working

### ISSUE THREAD

**Issue by wongwsvincent** [PennyLaneAI/pennylane#1330]
#### Issue description

The ```ranges``` argument of ```qml.templates.layers.StronglyEntanglingLayers``` cannot be set by the user, as there is a bug in the code (```self.ranges``` is not defined if the argument ```ranges``` is not None)

* *Expected behavior:* ```ranges``` can be defined by the user. The range value should be checked to ensure that it's not zero or divisible by the number of wires. The code should also be able to handle the range sequence if it has a length different from the number of layers. For example, it takes only the first ![formula](https://render.githubusercontent.com/render/math?math=\ell) range values if the range sequence length ![formula](https://render.githubusercontent.com/render/math?math=r) is longer than the number of layers ![formula](https://render.githubusercontent.com/render/math?math=\ell), and it repeats the range sequence if the range sequence length is shorter than the number of layers.

* *Actual behavior:* The following error message is returned: ```AttributeError: 'StronglyEntanglingLayers' object has no attribute 'ranges'```

* *Reproduces how often:* Always

* *System information:* (post the output of `import pennylane as qml; qml.about()`)
```
Name: PennyLane
Version: 0.16.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /Users/vwong/Documents/Projects/pennylane
Requires: numpy, scipy, networkx, autograd, toml, appdirs, semantic-version
Required-by: 
Platform info:           Darwin-18.7.0-x86_64-i386-64bit
Python version:          3.7.10
Numpy version:           1.19.5
Scipy version:           1.6.3
Installed devices:
- default.gaussian (PennyLane-0.16.0.dev0)
- default.mixed (PennyLane-0.16.0.dev0)
- default.qubit (PennyLane-0.16.0.dev0)
- default.qubit.autograd (PennyLane-0.16.0.dev0)
- default.qubit.jax (PennyLane-0.16.0.dev0)
- default.qubit.tf (PennyLane-0.16.0.dev0)
- default.tensor (PennyLane-0.16.0.dev0)
- default.tensor.tf (PennyLane-0.16.0.dev0)
```

#### Source code

Relevant lines of ```StronglyEntanglingLayers``` source code:
https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/templates/layers/strongly_entangling.py#L89

**Comment by wongwsvincent** [comment]
I am happy to submit a PR for the fix if everyone agrees that the expected behavior described above is how we should implement it.

**Comment by josh146** [comment]
Hi @wongwsvincent! Thanks for making a separate issue for this. 

> I am happy to submit a PR for the fix if everyone agrees that the expected behavior described above is how we should implement it.

Your solution above makes sense, a PR with a fix would be very appreciated!
